### PR TITLE
Schema model improvements

### PR DESCRIPTION
**Pull request by MacondoExpress**
# Description

- Removes isListOf, 
- Add assertionOptions to the Attribute Type assertions,
- Sync with the translation layer Relationship/RelationshipAdapter
- Removes Point and CartesianPoint from Scalars.
- Removes the Scalar "category "field, now inferred in the adapters.

## Complexity

Complexity: Low

Closes 

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [x] Documentation has been updated
- [x] TCK tests have been updated
- [x] Integration tests have been updated
- [x] Example applications have been updated
- [x] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the Neo4j GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-3794.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

**Comment by neo4j-team-graphql** [comment]
# Performance Report

No Performance Changes


<details>
    <summary>Show Full Table</summary>
    
| name | dbHits | old dbHits | time (ms) | old time (ms) | maxRows |
| ---- | ------ | ---------- | --------- | ------------- | ------- |
|  aggregations.TopLevelAggregate | 3403 | 3403 | 42 | 48 | 1134 |
|  aggregations.NestedAggregation | 16553 | 16553 | 73 | 133 | 2174 |
|  aggregations.AggregationWithWhere | 11979 | 11979 | 55 | 67 | 2174 |
|  aggregations.AggregationWhereWithinNestedRelationships | 22116987 | 22116987 | 3475 | 3319 | 2008534 |
|  aggregations.AggregationWhereWithinNestedConnections | 22116987 | 22116987 | 3257 | 3337 | 2008534 |
|  aggregations.NestedCountFromMovieToActors | 9734 | 9734 | 47 | 48 | 2174 |
|  aggregations.NestedCountFromActorsToMovie | 9937 | 9937 | 51 | 71 | 2174 |
|  aggregations.DeeplyNestedCount | 13070329 | 13070331 | 4923 | 5250 | 2008534 |
|  batch-create.BatchCreate | 4200 | 4200 | 215 | 258 | 600 |
|  batch-create.BatchCreateSmall | 77 | 77 | 87 | 101 | 11 |
|  connect.createAndConnect | 14424 | 14424 | 159 | 182 | 3003 |
|  connections.Connection | 14082 | 14082 | 73 | 123 | 2174 |
|  connections.NestedConnection | 45472 | 45468 | 118 | 133 | 4516 |
|  connections.ConnectionWithSort | 3284 | 3284 | 70 | 86 | 1040 |
|  connections.ConnectionWithSortAndCypher | 16325 | 16325 | 102 | 167 | 2174 |
|  create.SimpleMutation | 7 | 7 | 62 | 68 | 1 |
|  cypher-directive.TopLevelMutationDirective | 1135 | 1135 | 42 | 30 | 1134 |
|  delete.SimpleDelete | 19401 | 19401 | 300 | 269 | 1040 |
|  delete.NestedDeleteInUpdate | 19198 | 19198 | 209 | 185 | 1183 |
|  2925.SingleRelationshipFilter | 6468 | 6468 | 54 | 115 | 1040 |
|  2925.NestedSingleRelationshipFilter | 22900 | 22900 | 91 | 164 | 2174 |
|  2925.SingleRelationshipRequiredFilter | 5201 | 5201 | 41 | 51 | 1040 |
|  2925.NestedSingleRelationshipRequiredFilter | 9361 | 9361 | 62 | 87 | 1040 |
|  query.SimpleQuery | 3121 | 3121 | 25 | 33 | 1040 |
|  query.SimpleQueryWithRelationship | 16162 | 16162 | 51 | 67 | 2174 |
|  query.QueryWhere | 9718 | 9713 | 38 | 51 | 2168 |
|  query.SimpleQueryWithNestedWhere | 9896 | 9891 | 56 | 67 | 2168 |
|  query.Nested | 10096041 | 10096041 | 10770 | 10584 | 2008534 |
|  query.NestedWithFilter | 10074401 | 10074401 | 10617 | 10360 | 2004000 |
|  query.OrFilterOnRelationships | 41749 | 42681 | 235 | 218 | 1331 |
|  query.OrFilterOnRelationshipsAndNested | 34968 | 36326 | 234 | 298 | 1331 |
|  query.QueryWithNestedIn | 13596 | 13248 | 58 | 148 | 1283 |
|  query.NestedConnectionWhere | 9834 | 9834 | 73 | 143 | 2174 |
|  query.DeeplyNestedConnectionWhere | 9846 | 9926 | 91 | 185 | 2174 |
|  query.DeeplyNestedWithRelationshipFilters | 19026 | 19110 | 146 | 323 | 1558 |
|  query.NestedWithRelationshipSingleFilters | 3881 | 3881 | 183 | 502 | 1134 |
|  query.Fulltext | 80 | 80 | 36 | 72 | 16 |
|  query.FulltextWithNestedQuery | 587 | 587 | 47 | 85 | 84 |
|  sorting.SortMultipleTypes | 3515 | 3515 | 72 | 98 | 1040 |
|  sorting.SortMultipleTypesWithCypherWithCypher | 14526 | 14503 | 99 | 124 | 2174 |
|  sorting.SortOnNestedFields | 14082 | 14082 | 55 | 64 | 2174 |
|  sorting.SortDeeplyNestedFields | 43198 | 43198 | 99 | 136 | 4516 |
|  sorting.SortWithTopLevelCypher | 16162 | 16162 | 57 | 78 | 2174 |
|  unions.SimpleUnionQuery | 321 | 321 | 79 | 73 | 35 |
|  unions.SimpleUnionQueryWithMissingFields | 293 | 293 | 91 | 97 | 35 |
|  unions.NestedUnion | 410637 | 410637 | 393 | 300 | 33033 |
|  unions.NestedUnionWithMissingFields | 384611 | 384611 | 277 | 310 | 33033 |
|  update.NestedUpdate | 16143 | 16143 | 104 | 116 | 2002 |

</details>
        
Old Schema Generation: 45.610s
Schema Generation: 45.292s
Old Subgraph Schema Generation: 54.865s
Subgraph Schema Generation: 55.197s

File: packages/graphql/src/schema-model/attribute/model-adapters/AttributeAdapter.ts [hunk 1]
```diff
@@ -46,12 +47,17 @@ export class AttributeAdapter {
...
-
+    private assertionOptions: {
+        includeLists: boolean;
+    };
     constructor(attribute: Attribute) {
         this.name = attribute.name;
         this.type = attribute.type;
         this.annotations = attribute.annotations;
         this.databaseName = attribute.databaseName;
+        this.assertionOptions = {
```
Comment (a-alle) [inline packages/graphql/src/schema-model/attribute/model-adapters/AttributeAdapter.ts:58]: I still think this should be a default behaviour :D

**Review by a-alle — APPROVED** [review]
👍

**Pull request merged**